### ISSUE TITLE
Improve event updating to support retroactively identifying groups or updating group types

### ISSUE THREAD

**Issue by camerondeleone** [PostHog/posthog#11224]
## Is your feature request related to a problem?

It's not currently possible for users to update events to add properties, because it's expensive. However, a very common onboarding path is to set up tracking, collect data before identifying users or groups, and then have a bunch of unidentified data. Given the way distinct ids/person identification work, it's still possible to associate many of those events with a user if you then instrument the proper `identify` calls and they are still seen with the same `distinct_id`. There is no equivalent for groups, though. If a user sends a bunch of data with no groups, there's no way to retroactively add groups back to events. But this is a common experience of onboarding to PostHog — to set groups up after using the platform for a while and finding its value. 

In the simplest case (a 1:1, unchanging relationship between person and a group, such as with company/organization membership), this feels frustrating. In other uses of groups (such as when an end user interacts with different vendors on a marketplace), it's understandably more complicated for the user to pass this information back in. I am specifically trying to solve for the former here (ie, the simpler 1:1 durable relationship case). 

## Describe the solution you'd like

The specific request was a patch endpoint for events, where group info could be added, but this might take other forms to achieve the same thing.

As an aside, this type of operation could also unblock the ability to rename group types (though we probably want to be careful with exposing that because it seems trivial to the end user but could be really expensive depending on rows affected)

## Describe alternatives you've considered

Do nothing and tell users they cannot update or retroactively identify groups

Alternative path for users is to start a new project, migrate all data using migrator, write a custom plugin to add logic to associate events with groups, and write all updated data using it to new project. Not ideal,

**Comment by camerondeleone** [comment]
Related to #11128

**Comment by joepavitt** [comment]
Assuming, given this is stale, no updates on this? Have the same request from us atm, would be super helpful, but appreciate how expensive each transaction would be here

**Comment by camerondeleone** [comment]
@joepavitt I am closing this as won't fix. We now expose the underlying schema for groups, persons, and events so if you have a one off analysis you can do this at query time using a sql insight, but it's not something we're going to tackle for the time being otherwise.

**Comment by joepavitt** [comment]
Thanks for the update @camerondeleone - could I trouble you for an example QL please? Say we want to get all historical pageviews, grouped by "Group 1" (our "Team" group)? The events are not tagged with those events, but the "Person" would be. It hasn't been clear in the QL how to get the equivalent of the "Related Groups" view/list for a given "Person"?
